store/sqlstore/channel_store.go

```diff
@@ -3496,6 +3496,38 @@ func (s SqlChannelStore) UserBelongsToChannels(userId string, channelIds []strin
 	return c > 0, nil
 }
 
+func (s SqlChannelStore) GetDirectChannelsForUser(userId string) ([]*model.Channel, error) {
+	query := `
+			SELECT
+				C.*,
+				OtherUsers.Username as DisplayName
+			FROM
+				Channels AS C
+			JOIN
+				ChannelMembers AS CM ON CM.ChannelId = C.Id
+			INNER JOIN (
+				SELECT
+					ICM.ChannelId AS ChannelId, IU.Username AS Username
+				FROM
+					Users as IU
+				JOIN
+					ChannelMembers AS ICM ON ICM.UserId = IU.Id
+				WHERE
+					IU.Id != :UserId
+				) AS OtherUsers ON OtherUsers.ChannelId = C.Id
+			WHERE
+			    C.Type = 'D'
+				AND CM.UserId = :UserId`
+
+	var channels model.ChannelList
+
+	if _, err := s.GetReplica().Select(&channels, query, map[string]interface{}{"UserId": userId}); err != nil {
+		return nil, model.NewAppError("SqlChannelStore.GetDirectChannelsForUser", "store.sql_channel.direct_channels_for_user.app_error", nil, err.Error(), http.StatusInternalServerError)
```

Comment:
Since this adds a new error id, you'll need to run `make i18n-extract` and add a translation in `i18n/en.json`

Reply:
Looks like we're moving away from AppError, instead directly using the `error` type.

I'll defer to @agnivade  on this.

Reply:
yes, @agnivade in his previous comments mentioned switching to `error`